docs/configuration.md

```diff
@@ -1159,6 +1159,36 @@ combination with PAPERLESS_CONSUMER_BARCODE_UPSCALE bigger than 1.0.
 
     Defaults to "300"
 
+#### [`PAPERLESS_CONSUMER_ENABLE_TAG_BARCODE=<bool>`](#PAPERLESS_CONSUMER_ENABLE_TAG_BARCODE) {#PAPERLESS_CONSUMER_ENABLE_TAG_BARCODE}
+
+: Enables the detection of barcodes in the scanned document and
+assigns or creates tags if a properly formatted barcode is detected.
+
+    The barcode must match one of the (configurable) regular expressions.
+    If the barcode text contains ',' (comma), it is split into multiple
+    barcodes which are individually processed for tagging.
+
+    Matching is case insensitive.
+
+    Defaults to false.
+
+#### [`CONSUMER_TAG_BARCODE_MAPPING=<json dict>`](#CONSUMER_TAG_BARCODE_MAPPING) {#CONSUMER_TAG_BARCODE_MAPPING}
+
+: Defines a dictionary of filter regex and substitute expressions.
+
+    A barcode is only considered for tagging if at least one regex is matching
+    the barcode text. Before looking up or creating a tag, the substitute
+    is applied.
```

Comment:
```suggestion
    the barcode text. If a specified tag is not found it will be created.
```

I find saying "the substitute is applied" a bit confusing, I think it's better to describe this in terms of the user experience, which is to just find the regex and apply or create tags listed. Right?

Reply:
Agree, the whole paragraph was hard to understand. I rephrased it completely and added more examples.